### PR TITLE
Parallel building

### PR DESCRIPTION
**Pull request by The-going**
#### Description
Add parallel assembly relying on the correct operation of automatic dh commands.
- export all necessary variables and pass path to src dir
- fix override_dh_auto_clean
- PYTHON: Export a variable explicitly to access it
- dh_auto_build: Provide the possibility of automatic operation

close #2579

### CODE REVIEW

**Comment by The-going** [comment]
For debian buster 
https://github.com/LinuxCNC/linuxcnc/actions/runs/5681564341/job/15398017616?pr=2599#step:7:4112
--libdir=\${prefix}/lib/x86_64-linux-gnu
and
https://github.com/LinuxCNC/linuxcnc/actions/runs/5681564341/job/15398017616?pr=2599#step:7:6682
```
dh_install: linuxcnc-uspace-dev missing files: usr/lib/liblinuxcnc.a
dh_install: Cannot find (any matches for) "usr/lib/*.so" (tried in ., debian/tmp)
```
Now these paths
https://github.com/LinuxCNC/linuxcnc/blob/0536a28cdc52c99951a23be06e742692231cdab0/debian/linuxcnc.install.in#L113-L114
should be
```
usr/lib/${DEB_HOST_GNU_TYPE}/lib*.so.*
usr/lib/${DEB_HOST_GNU_TYPE}/linuxcnc
```

**Comment by The-going** [comment]
Parallel build completed successfully.

 package-arch (debian:buster)
succeeded Jul 28, 2023 in 23m 15s

The work within this pull request is completed.

File: debian/rules.in [hunk 1]
```diff
@@ -31,40 +33,41 @@ export TIME=$(shell LANG=C date --date='@$(TIMESTAMP)' '+%T')
...
-	dh $@
+	dh $@ -D$(SRCDIR)
 
 override_dh_auto_configure:
 	cd src && ./autogen.sh
-	cd src && PYTHON=/usr/bin/python3 ./configure \
+	dh_auto_configure -- \
 	    --prefix=/usr --sysconfdir=/etc \
+	    --libdir=/usr/lib \
+	    --disable-multiarch \
```
Comment (petterreinholdtsen) [inline debian/rules.in:50]: Are you sure you want to disable multiarch?

**Comment by petterreinholdtsen** [comment]
> usr/lib/${DEB_HOST_GNU_TYPE}/lib*.so.*

A simpler approach would be to just list /usr/lib/*//lib*.so.*.  Is there some reason this do not work on all versions and archs?  It seem like a better approach than dropping multiarch support.

**Comment by petterreinholdtsen** [comment]
Perhaps this change should aim for the 2.9 branch instead of master?  I suspect it would be better to avoid the white space changes to make it easier to see what is relevant for this change, and applied those independently to the 2.9 branch in f6a694297ab049162ce3ade3e3b6550b7179f6d5.

File: debian/rules.in (hunk 1 above)
Comment (The-going) [reply debian/rules.in:50]: I don't want to turn it off.
Changing library paths requires a thorough analysis of the source code that searches for these libraries or
sets this path at the very beginning when starting on a running system.

In addition, there are explicitly prescribed paths for x86:
```bash
linuxcnc> grep -nr 'lib/x86' ./*
....
./lib/python/qtvcp/designer/install_script:85:sopath=/usr/lib/x86_64-linux-gnu/qt5/plugins/designer/
./lib/python/qtvcp/designer/install_script:98:        echo -e '/usr/lib/x86_64-linux-gnu/qt5/plugins/designer/'
./scripts/setup_designer.in:85:sopath=/usr/lib/x86_64-linux-gnu/qt5/plugins/designer/
./scripts/setup_designer.in:98:        echo -e '/usr/lib/x86_64-linux-gnu/qt5/plugins/designer/'
....
linuxcnc> grep -nr '/lib' ./scripts/*
./scripts/linuxcnc.in:79:           LD_LIBRARY_PATH=$LINUXCNC_HOME/lib
./scripts/linuxcnc.in:81:           LD_LIBRARY_PATH=$LINUXCNC_HOME/lib:"$LD_LIBRARY_PATH"
.....
./scripts/runtests.in:28:    export LIBDIR=${TOPDIR}/lib
./scripts/runtests.in:35:    export LIBDIR=@EMC2_HOME@/lib
```
And this needs to be fixed in any case.

We will reverse this commit in the next pull request when I analyze and
test it on running hardware that is different from x86

**Comment by The-going** [comment]
> A simpler approach would be to just list

I completely agree with you here. A simple template like:
`usr/lib/*/lib*.so.*`
the right solution.

**Comment by The-going** [comment]
> Perhaps this change should aim for the 2.9 branch instead of master? I suspect it would be better to avoid the white space changes to make it easier to see what is relevant for this change, and applied those independently to the 2.9 branch in [f6a6942](https://github.com/LinuxCNC/linuxcnc/commit/f6a694297ab049162ce3ade3e3b6550b7179f6d5).

I didn't notice that the 2.8 branch is merged into 2.9 and 2.9 is merged into the master.
Thus, the changes made only in the wizard do not fall into 2.8, 2.9.
Did I understand correctly?

**Comment by The-going** [comment]
> Perhaps this change should aim for the 2.9 branch instead of master?

Done. https://github.com/LinuxCNC/linuxcnc/pull/2603

**Comment by petterreinholdtsen** [comment]
Replaced by #2603.

**Pull request closed without merging**